File: Extension Objects/Full Extensions/_Source Codes/MMF2 SDK/iOS Haptic Feedback/iOSHaptics - Source for iOS SDK/CExtLoad.m
//----------------------------------------------------------------------------------
//
// CEXTLOAD Chargement des extensions
//
//----------------------------------------------------------------------------------
#import "CExtLoad.h"
#import "CFile.h"
#import "CRunExtension.h"

//F01
#import "CRunKcArray.h"
#import "CRunActiveBackdrop.h"
#import "CRunAdvGameBoard.h"
#import "CRunAdvPathMov.h"
#import "CRunCalcRect.h"
#import "CRunIIF.h"
#import "CRunInAndOutController.h"
#import "CRunkcdirect.h"
#import "CRunLayer.h"
#import "CRunMoveSafely2.h"
#import "CRunPlatform.h"
#import "CRunclickteam_movement_controller.h"
#import "CRunKcBoxA.h"
#import "CRunKcBoxB.h"
#import "CRunAccelerometer.h"
#import "CRunLocation.h"
#import "CRunkcini.h"
#import "CRunMultipleTouch.h"
#import "CRuniOSButton.h"
#import "CRuniOSSingleEdit.h"
#import "CRuniOSMultipleEdit.h"
#import "CRunJoystickControl.h"
#import "CRunAdvDir.h"
#import "CRunkcclock.h"
#import "CRunkchisc.h"
#import "CRunkcrandom.h"
#import "CRunObjectMover.h"
#import "CRunWargameMap.h"
#import "CRunparser.h"
#import "CRunMTRandom.h"
#import "CRunGet.h"
#import "CRunkclist.h"
#import "CRunkcinput.h"
#import "CRuniOSStore.h"
#import "CRunStringTokenizer.h"
#import "CRunMoveIt.h"
#import "CRunEasing.h"
#import "CRunkcedit.h"
#import "CRuniOS.h"
#import "CRunkcbutton.h"
#import "CRunCamera.h"
#import "CRunGameCenterConnect.h"
#import "CRunGameCenterLeaderboard.h"
#import "CRunGameCenterAchievements.h"
#import "CRunGameCenterAchievement.h"
#import "CRunGameCenterMultiplayer.h"

// My extensions
#import "CRuniOSHaptics.h"

//F01END

@implementation CExtLoad

-(void)dealloc
{
	[name release];
	[subType release];
	[super dealloc];
}
-(void)loadInfo:(CFile*)file
{
	int debut = [file getFilePointer];
	
	short size = [file readAShort];
	handle = [file readAShort];
	[file skipBytes:12];
	
	name = [file readAString];
	NSRange index = [name rangeOfString:@"."];
	if (index.location!=NSNotFound)
	{
		NSString* temp=[name substringToIndex:index.location];
		[name release];
		name=temp;
		[name retain];
	}
	subType = [file readAString];
	
	[file seek:debut + size];
}

-(CRunExtension*)loadRunObject 
{
	CRunExtension* object=nil;
	
//F02 			
	if ([name caseInsensitiveCompare:@"kcarray"]==0)
	{
		object=[[CRunKcArray alloc] init];
	}
	if ([name caseInsensitiveCompare:@"ActiveBackdrop"]==0)
	{
		object=[[CRunActiveBackdrop alloc] init];
	}
	if ([name caseInsensitiveCompare:@"AdvGameBoard"]==0)
	{
		object=[[CRunAdvGameBoard alloc] init];
	}
	if ([name caseInsensitiveCompare:@"AdvPathMov"]==0)
	{
		object=[[CRunAdvPathMov alloc] init];
	}
	if ([name caseInsensitiveCompare:@"CalcRect"]==0)
	{
		object=[[CRunCalcRect alloc] init];
	}
	if ([name caseInsensitiveCompare:@"IIF"]==0)
	{
		object=[[CRunIIF alloc] init];
	}
	if ([name caseInsensitiveCompare:@"InAndOutController"]==0)
	{
		object=[[CRunInAndOutController alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcdirect"]==0)
	{
		object=[[CRunkcdirect alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Layer"]==0)
	{
		object=[[CRunLayer alloc] init];
	}
	if ([name caseInsensitiveCompare:@"MoveSafely2"]==0)
	{
		object=[[CRunMoveSafely2 alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Platform"]==0)
	{
		object=[[CRunPlatform alloc] init];
	}
	if ([name caseInsensitiveCompare:@"clickteam-movement-controller"]==0)
	{
		object=[[CRunclickteam_movement_controller alloc] init];
	}
	if ([name caseInsensitiveCompare:@"KcBoxA"]==0)
	{
		object=[[CRunKcBoxA alloc] init];
	}
	if ([name caseInsensitiveCompare:@"KcBoxB"]==0)
	{
		object=[[CRunKcBoxB alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Accelerometer"]==0)
	{
		object=[[CRunAccelerometer alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Location"]==0)
	{
		object=[[CRunLocation alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcini"]==0)
	{
		object=[[CRunkcini alloc] init];
	}
	if ([name caseInsensitiveCompare:@"MultipleTouch"]==0)
	{
		object=[[CRunMultipleTouch alloc] init];
	}
	if ([name caseInsensitiveCompare:@"iOSButton"]==0)
	{
		object=[[CRuniOSButton alloc] init];
	}
	if ([name caseInsensitiveCompare:@"iOSSingleEdit"]==0)
	{
		object=[[CRuniOSSingleEdit alloc] init];
	}
	if ([name caseInsensitiveCompare:@"iOSMultipleEdit"]==0)
	{
		object=[[CRuniOSMultipleEdit alloc] init];
	}
	if ([name caseInsensitiveCompare:@"AdvDir"]==0)
	{
		object=[[CRunAdvDir alloc] init];
	}
	if ([name caseInsensitiveCompare:@"JoystickControl"]==0)
	{
		object=[[CRunJoystickControl alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcclock"]==0)
	{
		object=[[CRunkcclock alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kchisc"]==0)
	{
		object=[[CRunkchisc alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcrandom"]==0)
	{
		object=[[CRunkcrandom alloc] init];
	}
	if ([name caseInsensitiveCompare:@"ObjectMover"]==0)
	{
		object=[[CRunObjectMover alloc] init];
	}
	if ([name caseInsensitiveCompare:@"WargameMap"]==0)
	{
		object=[[CRunWargameMap alloc] init];
	}
	if ([name caseInsensitiveCompare:@"parse"]==0)
	{
		object=[[CRunparser alloc] init];
	}
	if ([name caseInsensitiveCompare:@"MTRandom"]==0)
	{
		object=[[CRunMTRandom alloc] init];
	}
	if ([name caseInsensitiveCompare:@"parser"]==0)
	{
		object=[[CRunparser alloc] init];
	}
	if ([name caseInsensitiveCompare:@"get"]==0)
	{
		object=[[CRunGet alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kclist"]==0)
	{
		object=[[CRunkclist alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcinput"]==0)
	{
		object=[[CRunkcinput alloc] init];
	}
	if ([name caseInsensitiveCompare:@"iOSStore"]==0)
	{
		object=[[CRuniOSStore alloc] init];
	}	
	if ([name caseInsensitiveCompare:@"StringTokenizer"]==0)
	{
		object=[[CRunStringTokenizer alloc] init];
	}
	if ([name caseInsensitiveCompare:@"MoveIt"]==0)
	{
		object=[[CRunMoveIt alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Easing"]==0)
	{
		object=[[CRunEasing alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcedit"]==0)
	{
		object=[[CRunkcedit alloc] init];
	}
	if ([name caseInsensitiveCompare:@"iOS"]==0)
	{
		object=[[CRuniOS alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcbutton"]==0)
	{
		object=[[CRunkcbutton alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Camera"]==0)
	{
		object=[[CRunCamera alloc] init];
	}
	if ([name caseInsensitiveCompare:@"GameCenterConnect"]==0)
	{
		object=[[CRunGameCenterConnect alloc] init];
	}
	if ([name caseInsensitiveCompare:@"GameCenterLeaderboard"]==0)
	{
		object=[[CRunGameCenterLeaderboard alloc] init];
	}
	if ([name caseInsensitiveCompare:@"GameCenterAchievements"]==0)
	{
		object=[[CRunGameCenterAchievements alloc] init];
	}
	if ([name caseInsensitiveCompare:@"GameCenterAchievement"]==0)
	{
		object=[[CRunGameCenterAchievement alloc] init];
	}
	if ([name caseInsensitiveCompare:@"GameCenterMultiplayer"]==0)
	{
		object=[[CRunGameCenterMultiplayer alloc] init];
	}

	// My extensions
	if ([name caseInsensitiveCompare:@"iOSHaptics"]==0)
	{
		object=[[CRuniOSHaptics alloc] init];
	}
	
	//F02END	
	
	return object;
}

@end

File: Extension Objects/Runtime Ports/_Source Codes (XNA & Java)/XNA/CExtLoad.cs
//----------------------------------------------------------------------------------
//
// CEXTLOADER: Chargement des extensions / Loading extensions
//
//----------------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using RuntimeXNA.Services;

namespace RuntimeXNA.Extensions
{
    class CExtLoad
    {
        public string name;
        public string subType;
        public short handle;

        public void loadInfo(CFile file)
        {
            int debut = file.getFilePointer();

            short size = file.readAShort();
            handle = file.readAShort();
            file.skipBytes(12);
            name = file.readAString();
            int pos = name.LastIndexOf('.');
            name = name.Substring(0, pos);
            subType = file.readAString();
            file.seek(debut + size);
        }

        public CRunExtension loadRunObject()
        {
	        CRunExtension pObject=null;
        	
//F01 			
	        if (string.Compare(name, "kcarray")==0)
	        {
		        pObject=new CRunKcArray();
	        }
            if (string.Compare(name, "KcDbl") == 0)
            {
                pObject = new CRunKcDbl();
            }
            if (string.Compare(name, "ActiveBackdrop")==0)
	        {
		        pObject=new CRunActiveBackdrop();
	        }
	        if (string.Compare(name, "AdvGameBoard")==0)
	        {
		        pObject=new CRunAdvGameBoard();
	        }
	        if (string.Compare(name, "AdvPathMov")==0)
	        {
		        pObject=new CRunAdvPathMov();
	        }
//	        if (string.Compare(name, "CalcRect")==0)
//	        {
//		        pObject=new CRunCalcRect();
//	        }
	        if (string.Compare(name, "IIF")==0)
	        {
		        pObject=new CRunIIF();
	        }
	        if (string.Compare(name, "InAndOutController")==0)
	        {
		        pObject=new CRunInAndOutController();
	        }
	        if (string.Compare(name, "kcdirect")==0)
	        {
		        pObject=new CRunkcdirect();
	        }
	        if (string.Compare(name, "Layer")==0)
	        {
		        pObject=new CRunLayer();
	        }
	        if (string.Compare(name, "MoveSafely2")==0)
	        {
		        pObject=new CRunMoveSafely2();
	        }
	        if (string.Compare(name, "Platform")==0)
	        {
		        pObject=new CRunPlatform();
	        }
	        if (string.Compare(name, "clickteam-movement-controller")==0)
	        {
		        pObject=new CRunclickteam_movement_controller();
	        }
	        if (string.Compare(name, "KcBoxA")==0)
	        {
		        pObject=new CRunKcBoxA();
	        }
	        if (string.Compare(name, "KcBoxB")==0)
	        {
		        pObject=new CRunKcBoxB();
	        }
//	        if (string.Compare(name, "Accelerometer")==0)
//	        {
//		        pObject=new CRunAccelerometer();
//	        }
//	        if (string.Compare(name, "Location")==0)
//	        {
//		        pObject=new CRunLocation();
//	        }
	        if (string.Compare(name, "kcini")==0)
	        {
		        pObject=new CRunkcini();
	        }
//	        if (string.Compare(name, "MultipleTouch")==0)
//	        {
//		        pObject=new CRunMultipleTouch();
//	        }
	        if (string.Compare(name, "KcButton")==0)
	        {
		        pObject=new CRunKcButton();
	        }
//	        if (string.Compare(name, "iOSSingleEdit")==0)
//	        {
//		        pObject=new CRuniOSSingleEdit();
//	        }
//	        if (string.Compare(name, "iOSMultipleEdit")==0)
//	        {
//		        pObject=new CRuniOSMultipleEdit();
//	        }
	        if (string.Compare(name, "AdvDir")==0)
	        {
		        pObject=new CRunAdvDir();
	        }
//	        if (string.Compare(name, "JoystickControl")==0)
//	        {
//		        pObject=new CRunJoystickControl();
//	        }
//	        if (string.Compare(name, "kcclock")==0)
//	        {
//		        pObject=new CRunkcclock();
//	        }
	        if (string.Compare(name, "kchisc")==0)
	        {
		        pObject=new CRunkchisc();
	        }
	        if (string.Compare(name, "kcrandom")==0)
	        {
		        pObject=new CRunkcrandom();
	        }
	        if (string.Compare(name, "ObjectMover")==0)
	        {
		        pObject=new CRunObjectMover();
	        }
	        if (string.Compare(name, "WargameMap")==0)
	        {
		        pObject=new CRunWargameMap();
	        }
	        if (string.Compare(name, "parser")==0)
	        {
		        pObject=new CRunparser();
	        }
	        if (string.Compare(name, "MTRandom")==0)
	        {
		        pObject=new CRunMTRandom();
	        }
	        if (string.Compare(name, "XNAEdit")==0)
	        {
		        pObject=new CRunXNAEdit();
	        }
//	        if (string.Compare(name, "get")==0)
//	        {
//		        pObject=new CRunGet();
//	        }
	        if (string.Compare(name, "kclist")==0)
	        {
		        pObject=new CRunkclist();
	        }
            if (string.Compare(name, "kccombo") == 0)
            {
                pObject = new CRunkccombo();
            }
            if (string.Compare(name, "kcedit") == 0)
            {
                pObject = new CRunkcedit();
            }
//	        if (string.Compare(name, "kcinput")==0)
//	        {
//		        pObject=new CRunkcinput();
//	        }
//	        if (string.Compare(name, "MobileUtilities")==0)
//	        {
//		        pObject=new CRunMobileUtilities();
//	        }
	        if (string.Compare(name, "StringTokenizer")==0)
	        {
		        pObject=new CRunStringTokenizer();
	        }
			// My extension ports
			if (string.Compare(name, "commentor")==0)
	        {
		        pObject=new CRuncommentor();
	        }
			if (string.Compare(name, "AdvComment")==0)
	        {
		        pObject=new CRunAdvComment();
	        }
			if (string.Compare(name, "OpenURLs")==0)
	        {
		        pObject=new CRunOpenURLs();
	        }
			if (string.Compare(name, "fmod")==0)
	        {
		        pObject=new CRunfmod();
	        }
			if (string.Compare(name, "ascii")==0)
	        {
		        pObject=new CRunascii();
	        }
			if (string.Compare(name, "AsciiChar")==0)
	        {
		        pObject=new CRunAsciiChar();
	        }
			if (string.Compare(name, "IntegerSplit")==0)
	        {
		        pObject=new CRunIntegerSplit();
	        }
			// ------------------
#if WINDOWS_PHONE
            if (string.Compare(name, "MultipleTouch") == 0)
            {
                pObject = new CRunMultipleTouch();
            }
            if (string.Compare(name, "JoystickControl") == 0)
            {
                pObject = new CRunJoystickControl();
            }
            if (string.Compare(name, "Accelerometer") == 0)
            {
                pObject = new CRunAccelerometer();
            }
            if (string.Compare(name, "Location") == 0)
            {
                pObject = new CRunLocation();
            }
#endif
//F01END	        	
	        return pObject;
        }
    }
}